privateSBOMExchange/src/petra/lib/models/__init__.py

```diff
@@ -207,7 +207,7 @@ def visit_sbom_node(self, node:SbomNode):
             The computed hash of the SBOM node data and its children.
         """
         children_hashes = b''.join(child.accept(self) for child in node.children)
-        data_to_hash = (node.purl).encode() + children_hashes
+        data_to_hash = (f"{node.complex_type}{node.purl}").encode() + children_hashes
```

Comment:
Would we want to use a better encoding for the field information (e.g., as a json, or so?) I worry that (perhaps not for complex nodes) for some types we end up with fields that appear as other fields when we de-serialize.

Reply:
I wouldn't want to hash json-encoded data, and I think it's actually a feature to obscure the nodes when the tree gets serialized. 

But you're right that we still want to be able to be able to reconstruct the tree correctly when we de-serialize. Here's the information per node we need to reconstruct: node type, encrypted_data, public_data, policy, hash, children/parents. So I can add a quick and dirty serialization/deserialization function to put trees on disk.